### ISSUE TITLE
Problems with python3

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#608]
(Original issue 901 created by None on 2013-09-06T19:25:03.611154+00:00)

pygments, `guess_lexer_for_filename` currently has problems with python3.

example python2:

```
#!python

from pygments.lexers import guess_lexer_for_filename
guess_lexer_for_filename('bla.php', '<?php')
#-> <pygments.lexers.PhpLexer>
```

example python3

```
#!python

from pygments.lexers import guess_lexer_for_filename
guess_lexer_for_filename('bla.php', '<?php')
```

```
#!text

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-31-40f0c3196d26> in <module>()
----> 1 guess_lexer_for_filename('bla.php', '<?php')

/usr/lib/python3.3/site-packages/pygments/lexers/__init__.py in guess_lexer_for_filename(_fn, _text, **options)
    188             return lexer(**options)
    189         result.append((rv, lexer))
--> 190     result.sort()
    191     if not result[-1][0] and primary is not None:
    192         return primary(**options)

TypeError: unorderable types: LexerMeta() < LexerMeta()
```

I am on arch linux using the package `python-pygments v.1.6`

**Comment by Anteru** [comment]
(Original issue 901 last updated on 2014-04-21T23:38:35.969575+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T19:04:05.464146+00:00)

Thanks, this is already fixed in tip (see c31841d2fd21)

**Comment by Anteru** [comment]
(Original comment by msabramo on 2014-04-21T23:38:35.964170+00:00)

Is there a PyPI release planned that will incorporate this fix?

Just wondering, because I noticed that folks are pinning and vendoring stuff to work around the problem:

https://github.com/edx/diff-cover/pull/65/files#r11827648
